File: MediGuide.Domain/Entities/Agent.cs
using MediGuide.Domain.Common;

namespace MediGuide.Domain.Entities;

public class Agent : BaseEntity
{
    public string FullName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string PhoneNumber { get; set; } = string.Empty;
    public bool IsAvailable { get; set; } = true;
    public bool IsActive { get; set; } = true;

    // Navigation
    public ICollection<Booking> AssignedBookings { get; set; } = new List<Booking>();
}

File: medi-guide-web/src/app/core/services/agent.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { environment } from '../../../environments/environment';

export interface AgentDto {
  id: string;
  fullName: string;
  email: string;
  phoneNumber: string;
  isAvailable: boolean;
  isActive: boolean;
}

export interface RegisterAgentRequest {
  fullName: string;
  email: string;
  phoneNumber: string;
  password: string;
}

@Injectable({ providedIn: 'root' })
export class AgentService {
  private readonly http = inject(HttpClient);
  private readonly api = environment.apiUrl;

  getAll() {
    return this.http.get<AgentDto[]>(`${this.api}/agents`);
  }

  register(dto: RegisterAgentRequest) {
    return this.http.post(`${this.api}/auth/register-agent`, dto);
  }
}